Datasnap/Classes/Utilities/Datasnap.m

```diff
@@ -147,23 +160,23 @@ - (void)onDataInitialized
     }
     switch (self.vendorProperties.vendor) {
     case GIMBAL:
-        self.gimbalClient = [[GimbalClient alloc] init];
-        self.gimbalClient.gimbalApiKey = self.vendorProperties.gimbalApiKey;
-        [self.gimbalClient startGimbal];
+        if ([AFNetworkReachabilityManager sharedManager].reachable) {
+            self.gimbalClient = [GimbalClient new];
+            self.gimbalClient.gimbalApiKey = self.vendorProperties.gimbalApiKey;
+            [self.gimbalClient startGimbal];
+        }
         break;
     case ESTIMOTE:
         break;
     }
-    if ([[NSUserDefaults standardUserDefaults] boolForKey:@"isAppAlreadyLaunchedOnce"]) {
-    }
-    else {
+    if (![[NSUserDefaults standardUserDefaults] boolForKey:@"isAppAlreadyLaunchedOnce"]) {
         [[NSUserDefaults standardUserDefaults] setBool:YES forKey:@"isAppAlreadyLaunchedOnce"];
         [[NSUserDefaults standardUserDefaults] synchronize];
         NSString* eventType = appInstalledEventType;
-        BaseEvent* event = [[BaseEvent alloc] init];
+        BaseEvent* event = [BaseEvent new];
         event.eventType = eventType;
-        [event.organizationIds addObject:self.organizationId];
-        [event.projectIds addObject:self.projectId];
+        event.organizationIds = @[ self.organizationId ];
```

Comment:
Same situation here as a comment below... This kind of thing should be in a custom init method for BaseEvent.
`BaseEvent *event = [[BaseEvent alloc] initWithType:eventType organizationId:organizationId projectId:projectId]`